### PR TITLE
Several improvements

### PR DESCRIPTION
**Pull request by tsagadar**
Several cleanups and code improvements. Best to review commit by commit.

### CODE REVIEW

File: tests/updater/controller/v1/test_routes.py
Comment (b-rowan) [inline tests/updater/controller/v1/test_routes.py:145]: There is a good chance this is to check the size of the firmware.  I think we can implement this for local files, can do some research into what a file header should send.

File: goosebit/updater/manager.py
Comment (b-rowan) [inline goosebit/updater/manager.py:68]: Do we care to rename these for the `configData` endpoint?  Is `revision` or `model` used anywhere else important?  If so I'm not opposed to renaming these to keep it consistent.

File: goosebit/updater/manager.py
Comment (b-rowan) [inline goosebit/updater/manager.py:213]: Is this something we would consider a warning?  I'm sure later that can be extracted to be much more clear for the user on the web UI, but logging it as a warning here for now might not be a bad start?

File: goosebit/models.py
Comment (b-rowan) [inline goosebit/models.py:28]: Same issue, can this just be a `StrEnum`?

File: goosebit/models.py
Comment (b-rowan) [inline goosebit/models.py:65]: Why `null=True` here?  I think `UNKNOWN` is just the same as `None`, so we should decide which one we want.  My tendency is to lean towards `None`, but I think there might be a case for `UNKNOWN` such as updating it with any variant of this when the state information is lost or unparsable for some reason, but I still think `None` is reasonable.

**Review by b-rowan — CHANGES_REQUESTED** [review]
A couple small changes and confirmations on things here, overall huge amount of good quality fixes, thanks!

File: goosebit/updater/manager.py
Comment (easybe) [reply goosebit/updater/manager.py:68]: Maybe @sbabic can help us come with a good convention for attributes (“config data”). What do others use?

File: goosebit/api/devices.py
Comment (tsagadar) [reply goosebit/api/devices.py:74]: Agree. Reverted.

File: goosebit/models.py
Comment (tsagadar) [reply goosebit/models.py:11]: `IntEnum` is more efficient in storage and faster, also has less potential for typos. Downside: database entries are not as human readable. Not sure what we should value more.

File: goosebit/updater/manager.py
Comment (tsagadar) [reply goosebit/updater/manager.py:193]: Did not think deeply about it when adding it. It seemed better than a comment and could be an early warning if we ever extend the enum and forgot to adjust the code.

In production, assertions can be disabled using the -O (optimize) flag - something I'd propose to do if we keep assertions.

But agree: if we only have a handful of such assertions we better eliminate them. Proposal: let's try to use them for a bit. If not many good usages come up, then remove them.

File: goosebit/updater/manager.py
Comment (tsagadar) [reply goosebit/updater/manager.py:280]: Yes - I am also not too happy with the global state but did not yet look deeper into it.

`force_update` likely could be replaced by setting the `update_state` of a device to `registered' or 'finished' and resetting device.fw_version. Maybe it then would be called "reset state".

File: goosebit/updater/manager.py
Comment (tsagadar) [reply goosebit/updater/manager.py:147]: Yes - it is correct (as the automated tests also verify). Still changed it to your proposal as it is a bit simpler.

File: tests/updater/controller/v1/test_routes.py
Comment (tsagadar) [reply tests/updater/controller/v1/test_routes.py:145]: You were spot on: looks like swupdate queries the size of the firmware. Implemented endpoint.

File: goosebit/updater/manager.py
Comment (tsagadar) [reply goosebit/updater/manager.py:213]: Agree. Changed.

File: goosebit/models.py
Comment (tsagadar) [reply goosebit/models.py:65]: Correct. Changed.

File: goosebit/updater/manager.py
Comment (tsagadar) [reply goosebit/updater/manager.py:280]: Note: there is currently also a bug when manually assigning a firmware to a device: the assigned firmware gets delete after a while - I think only in conjunction when the frontend is active.

Maybe another sign that we need to be careful with keeping cached data.

File: goosebit/models.py
Comment (b-rowan) [reply goosebit/models.py:11]: Ok, this is fine.

File: goosebit/updater/manager.py
Comment (b-rowan) [reply goosebit/updater/manager.py:193]: Sounds good, we can standardize using them where they may be applicable then.

File: goosebit/updater/manager.py
Comment (b-rowan) [reply goosebit/updater/manager.py:280]: > Yes - I am also not too happy with the global state but did not yet look deeper into it.
> 
> `force_update` likely could be replaced by setting the `update_state` of a device to `registered' or 'finished' and resetting device.fw_version. Maybe it then would be called "reset state".

This is not a bad idea, the idea of force update is that it ignores existing state of the device.  My only concern with this is that if the device sets that version afterwards it will remove the "force", which would happen on either update or reboot of a device.  Update is probably fine, but reboot isn't great.  

Maybe another idea is to extend `UpdateModeEnum` with a `FORCE` mode that always attempts to re-install the currently running firmware (both upgrades and downgrades should be able to be handled with existing flags), and it just sets it to pinned right after?

**Comment by tsagadar** [comment]
Implemented all of the feedback with the exception of the open discussions. Maybe adding issues for those that seem relevant?

Also did some more manual testing which lead to a few fixes and improvements. Once PR is merged I will create a few more issues with the open issues from my perspective.

**Comment by b-rowan** [comment]
> Implemented all of the feedback with the exception of the open discussions. Maybe adding issues for those that seem relevant?
> 
> 
> 
> Also did some more manual testing which lead to a few fixes and improvements. Once PR is merged I will create a few more issues with the open issues from my perspective.

Sounds good.  Should be clear to merge.

**Pull request merged**
